Lesson 4

SimreAniri opened the pull request.

SimreAniri committed:
Проведена оценка алгоритмов для поика количества кратных чисел.

SimreAniri committed:
Добавлен вывод по анализу алгоритмов.

SimreAniri committed:
Реализован поиск i-го простого числа с помощью решета Эратосфена.

SimreAniri committed:
Проведена оценка времени выполнения функции, использующей решето Эратосфена

SimreAniri committed:
Реализован алгоритм поиска i-ого простого числа.
Произведена оценка сложности алгоритма.

Dzhoker1 commented on Lesson_4_Task_1.py:
```diff
@@ -0,0 +1,138 @@
+# Рекурсий в алгоритме нет, время выполнения нарастает линейно.
+
+
+# ВЫВОД:
```
Математика в очередной раз победила информатику )))

Dzhoker1 commented on Lesson_4_Task_2.py:
```diff
@@ -0,0 +1,140 @@
+    return res[num - 1]
+
+
+def eratosthenes_sieve(n):
```
Мне столько вариантов сдавали, но ваш подход с просеиванием небольшими группами и использованием простых чисел в следующем просеивании я не встречал среди студенческих работ. 
Супер!

Dzhoker1 reviewed (commented):
Оставил комментарии по коду

SimreAniri committed:
В алгоритме с решетом Эратосфена при просеивании дополнительных групп при нахождении хотябы одного делителя среди простых чисел, дальнейшая проверка по простым числам не производится. Благодаря этому, временные показатели алгоритма улучшились.

В алгоритме поиска простых чисел простым перебором проверка происходит только для нечетных чисел. Временные показатели для входных данных до 1000 улучшелись незначительно.
